### ISSUE TITLE
Type Troubles with rtnetlink

### ISSUE THREAD

**Issue by skrap** [jbaublitz/neli#27]
Hello!

In trying to write some sample code for talking to the rtnetlink system, I ran into some difficulties using neli's type system.  A rtnetlink message's payload type depends on the `nlmsghdr`'s `nl_type` field.  A message to receive routing info (`Rtm::Getroute`) will receive a series of `Nlmsghdr<Nlmsg,Rtmsg<Rta>>`, followed by a single `Nlmsghdr<Nlmsg,NlEmpty>`.  This final message will have a `nl_type` of Nlmsg::Done.  

This looked pretty similar to what's in `ctrl-list.rs`, so I wrote some test code, based on `ctrl-list.rs`, with a core loop like so:
```
    let mut iter = socket.iter::<Nlmsg, Rtmsg<Rta>>();
    while let Some(response) = iter.next() {
       // stuff
    }
```
neli dutifully deserializes[1] all of the Rtmsg-based payloads, but fails on the Done message with `Error: failed to fill whole buffer` in `read_exact` because still tries to deserialize the `Rtmsg<Rta>` from the nonexistent payload bytes.

I don't exactly have a solution to these rough areas, but they were tricky to use, and it currently seems to me that I can't use the nice high-level `iter()` function, and instead will have to do raw `recv` calls.  Perhaps I'm missing something, though!  

[1] I have a bugfix for rtnl.rs — it seems that a `attr.strip(buf)` is needed to consume any unaligned padding bytes inside `Vec<Rtattr<T, P>>::deserialize`.  There's probably a companion padding bytes fix in serialize to be made.  Once I have something working I'll send something your way.

Do you have any advice or comments?  I will continue to try to use the low-level neli socket API to read the messages for the time being.

Thanks for all your great work on this!

**Comment by jbaublitz** [comment]
Hi @skrap, thanks for the feedback. If I'm reading correctly, I see two problems you're reporting here. One is the padding issue which I realized I had not fixed in `rtnl` right after releasing `0.4.3-r1`. I'll resolve that in `v0.4.3-r2` most likely with the help of your PR and I'll have you test before the actual release. The other is one that relates to issue #19 and the difficulty of parsing a number as one of a few enums all with the same marker trait. I'm going to update that issue with the proposed resolution tonight and feel free to add your feedback. As for your PR request, I'm about to hop over there to review that too. Thanks for the contributions!

**Comment by jbaublitz** [comment]
`master` has the associated PR merged. Closing.

If you have any other issues always feel free to let me know in #29. I will be reviewing rtnetlink support to see if I can improve it before v0.4.3 and it will be tracked using that issue. Thanks!
